### PR TITLE
Adding proposed structure for docs

### PR DESCRIPTION
**Pull request by benedictpaten**
This is the branch we are developing the docs for the paper on. Please @jvivian  @arkal @hannes-ucsc  @cket  @JakeNarkizian commit directly to this branch.

### CODE REVIEW

**Comment by arkal** [comment]
Jenkins, test this please.

**Comment by jvivian** [comment]
@benedictpaten — I've taken a first pass at all the sections assigned to me.  Let me know if there are any other sections you think we need or that I can do.

**Comment by hannes-ucsc** [comment]
@benedictpaten are you force pushing?

**Comment by hannes-ucsc** [comment]
@benedictpaten, ignore the previous comment. I'm trying to track down someone doing weird pushes.

**Comment by hannes-ucsc** [comment]
Before this can be merged, the "Enforce venv on sphinx-build" commit needs to be annotated with the issue it resolves, i.e. #752.

**Comment by jvivian** [comment]
@hannes-ucsc I'm getting errors now from conf.py, despite being in a venv.  Why is this extra check needed if the Makefile already checks for a virtualenv?

**Comment by cket** [comment]
The check is in conf.py because that is the only file guaranteed to run before building the docs. The makefile is bypassed by RTD, for example, by calling Sphinx-build directly. Are you on campus? I will be in the lab in about 15 minutes and I can take a look at the errors then

**Comment by jvivian** [comment]
@cket yup!

**Comment by cket** [comment]
Found the problem — the Toil venv was active but it did not contain the Sphinx installation. Installing Sphinx inside the venv fixed the issue.

File: docs/conf.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ def real_dir_name(p, n=1):
     print p
     return p
 
+if not hasattr(sys, 'real_prefix'):
+    raise RuntimeError('A virtualenv must be active')
```
Comment (hannes-ucsc) [inline docs/conf.py]: @cket, could you add ` and Sphinx must be installed in it.`

Comment (hannes-ucsc) [inline docs/conf.py]: This change would be in a separate PR against the documentation branch and it should use the same commit message as the original commit. When I squash down the doc branch, I'll merge those two commits.

**Comment by hannes-ucsc** [comment]
@cket, @jvivian I think we can avoid this confusion by making the exception message more informative. I commented on @cket's fbb35ed with details.

**Comment by benedictpaten** [comment]
Why are the tests flappy?

**Comment by jvivian** [comment]
> Why are the tests flappy?

Hannes aborted the last build due some jenkins thing he's fixing I think.

**Comment by hannes-ucsc** [comment]
I aborted a bunch of toil-branches/master builds, but no toil-pull-request builds.

**Comment by hannes-ucsc** [comment]
I aborted [this one](http://jenkins.cgcloud.info/job/toil-pull-requests/1050/) earlier today but that wasn't for this PR.

**Comment by hannes-ucsc** [comment]
> Look at the test numbers too, they are changing in between builds. Is that expected? 

The PR builder simulates a merge of the PR's source branch into the target branch and builds the resulting merge commit. If the merge fails, say, because of conflicts, the tip of the source branch is built instead. As long as there are no conflicts between source and target branch, you will observe changes on the target branch when the PR is built. When there are conflicts, those changes disappear again.

**Comment by benedictpaten** [comment]
I am going through all the documentation now doing an editing pass. I'd like to have this all merged today. Hannes, what commits do you want squashed? I'd be happy with merging as is, but then I never look at commit histories ;)

**Comment by jvivian** [comment]
@benedictpaten — Ping me if everything looks good after your pass, I already rendered most of the sections into their own PDF for the supplement and I'll have to rerender them if there are changes.

**Comment by jvivian** [comment]
@hannes-ucsc unfortunately has to deal with merge conflicts, there's too much conflict between @JakeNarkizian 's documentation changes from master and this branch.

**Comment by hannes-ucsc** [comment]
I'm taking off multi-author. Unless someone vetos until 4:00pm, I will be the owner of this branch and no one else is allowed to push to it. When I am done resolving the conflicts, we can open the branch up again for more commits or merge it.

**Comment by hannes-ucsc** [comment]
Continued in #778, rebased, partially squashed and with conflicts resolved.

**Pull request closed without merging**